torch/distributed/pipelining/schedules.py

```diff
@@ -583,6 +584,12 @@ def __init__(
         self._num_stages = stages[0].num_stages
         self.pp_group_size = stages[0].group_size
         self.rank = stages[0].group_rank
+        # Set the pipeline stage states
+        if stage_index_to_group_rank is not None:
+            for stage in self._stages:
+                stage.stage_index_to_group_rank = stage_index_to_group_rank
+        self.stage_index_to_group_rank = stages[0].stage_index_to_group_rank
```

Comment:
its a little weird to me to have the schedule set this inside the stage.  Does the stage need this information for creating ops? 

would it be cleaner if the schedule passed the 'dest rank' as an integer to stage.get_fwd_send_ops, for example?

Reply:
The stage does need this information since when it creates the `RecvInfo` classes to store the activations / grads (https://github.com/pytorch/pytorch/blob/main/torch/distributed/pipelining/stage.py#L52), it only stores the stage index.

When we actually create the ops it converts from `stage_index` -> `group_rank` (uses mapping) -> `global_rank` (uses distributed_c10d)